### PR TITLE
Update user management documentation after Okta migration

### PR DESCRIPTION
**Pull request by ikas**
This PR updates the documentation for user management after the migration to Okta has been completed. The changes done include:

- Update pagination for `GET users` endpoint, explaining the 2 supported pagination strategies (cursor and offset);
- Update callback endpoints for 3rd party login, since those are now handled by a single common endpoint;
- Removal of endpoints that are not supported anymore, since Okta takes care of those processes of user management for us;

### CODE REVIEW

File: source/includes/api/_user_management.md [hunk 1]
```diff
@@ -484,15 +371,34 @@ Error code     | Error message  | Description
...
+```shell
+curl -X GET "https://api.resourcewatch.org/auth/user?strategy=offset&page[number]=2&page[size]=25"
+-H "Content-Type: application/json"  -d \
+-H "Authorization: Bearer <your-token>" \
+```
+
+This endpoint currently supports 2 pagination strategies:
+
+* a strategy based on **limit** and **offset**, as you'll find with many other resources across the API. To use this strategy, you should provide a query parameter `strategy=offset` in your request, and identify the page size (`page[size]`) and page number (`page[number]`) you want to fetch. **This strategy has been deprecated - you should migrate to the cursor strategy defined below.**
+* a strategy based on **cursors** - for this strategy, you get the first page of the list, and then use the link defined in the `links.next` property oif the response body to fetch the following page. To use this strategy, you should provide a query parameter `strategy=cursor` in your request, and identify the cursor using the `page[after]` query parameter.
```
Comment (tiagojsag) [inline source/includes/api/_user_management.md]: oif -> of

File: source/includes/api/_user_management.md [hunk 2]
```diff
@@ -484,15 +371,34 @@ Error code     | Error message  | Description
...
+
+> DEPRECATED: Example request to load page 2 using 25 results per page using the "offset" strategy:
+
+```shell
+curl -X GET "https://api.resourcewatch.org/auth/user?strategy=offset&page[number]=2&page[size]=25"
+-H "Content-Type: application/json"  -d \
+-H "Authorization: Bearer <your-token>" \
+```
+
+This endpoint currently supports 2 pagination strategies:
```
Comment (tiagojsag) [inline source/includes/api/_user_management.md]: A few things missing here:
- what's used if you don't specify a strategy?
- the page based strategy should probably link to the existing docs on pagination
- On the cursor examples, could you add an example of the response that includes the new elements?
- What happens when you run out of data to load? (last "page" of the cursor)
- we should probably highlight that, at some point in the near future (do we have a date?), the page based pagination will be removed altogether

**Comment by tiagojsag** [comment]
Left some comments above. I'd like to also see some notes on the endpoints that may be indirectly affected by this - for example, we discussed the other day how okta cursors impact dataset/layer/widget listing that filters/sorts by user role. We should add a note on those docs to highlight that those may be slow, or ultimately not produce the expected results, due to the way okta works

File: source/includes/api/_user_management.md (hunk 2 above)
Comment (ikas) [reply source/includes/api/_user_management.md]: Thank you for the feedback :bow: I've detailed a bit more about the migration strategy itself, and about the 2 supported pagination strategies - I believe it now answers the questions you asked :+1:

**Comment by ikas** [comment]
Though I agree with you, the endpoints affected by that issue are only used internally by other services and are not documented. I've added a note to the "offset" strategy referring that the higher the page number, the slower the response will be.

**Comment by tiagojsag** [comment]
> Though I agree with you, the endpoints affected by that issue are only used internally by other services
I know, and I'd like to see those other services' endpoints documented

**Pull request merged**